packages/react-app/src/services/bounty.service.ts

```diff
@@ -83,34 +83,23 @@ export const getSortByValue = (originalInput: string): AcceptedSortOutputs => {
 	return output;
 };
 
-export const filterStatus = (query: FilterQuery<BountyCollection>, status?: string): FilterQuery<BountyCollection> => {
+export const filterStatus = (query: FilterQuery<BountyCollection>, status?: string[]): FilterQuery<BountyCollection> => {
 	/**
 	 * Pass status and append the corresponding status query to the query object
 	 */
-	if (status == null || status == '' || status == 'All' || status == undefined) {
-		query.status = { $in: ['Open', 'In-Progress', 'In-Review', 'Completed'] };
-	} else {
-		query.status = status;
-	}
+	
+	query.status = { $in: status && status.length ? status : ['Open', 'In-Progress', 'In-Review', 'Completed'] };
 	return query;
 };
 
-export const filterPaidStatus = (query: FilterQuery<BountyCollection>, paidStatus?: string): FilterQuery<BountyCollection> => {
+export const filterPaidStatus = (query: FilterQuery<BountyCollection>, paidStatus?: string[]): FilterQuery<BountyCollection> => {
 	/**
 	 * Pass paid status and append the corresponding status query to the query object
 	 */
-	if (paidStatus == null || paidStatus == '' || paidStatus == 'All' || paidStatus == undefined) {
-		query.$or = [
-			{ paidStatus: { $in: [PAID_STATUS.PAID, PAID_STATUS.UNPAID] } },
-			{ paidStatus: { $exists: false } },
-		];
-	} else if (paidStatus === PAID_STATUS.UNPAID) {
-		query.$or = [
-			{ paidStatus: PAID_STATUS.UNPAID },
-			{ paidStatus: { $exists: false } },
-		];
-	} else {
-		query.paidStatus = paidStatus;
+	query.$or = [{ paidStatus: { $in: paidStatus && paidStatus.length ? paidStatus : [PAID_STATUS.PAID, PAID_STATUS.UNPAID] } }];
```

Comment:
This is a lot more readable - nice!

though this brings up a usability thing to me:
`unpaid` and `paid` de-selected is equivalent to `unpaid` and `paid` selected. To us, I think this seems straightforward. Do you think that'll be the case for our users?

Reply:
I totally get your concern! I was struggling with the same initially. But after using it, contrary to my belief, I felt it was easy and intuitive. As a user, when the interpretation - when both were unselected, the filter is disabled - struck (pretty fast), it all made sense. I think the same logic is extendible to `status` as well.

Reply:
I think it will make sense to the users, or they will get it pretty quickly.